spec/langchain/assistant/assistant_spec.rb

```diff
@@ -482,6 +482,12 @@
         expect(subject.messages.first.content).to eq("New instructions")
         expect(subject.instructions).to eq("New instructions")
       end
+
+      it "clears instructions when instructions are nil" do
+        subject.instructions = nil
+        expect(subject.messages).to be_empty
+        expect(subject.instructions).to be_nil
```

Comment:
Can we enhance this spec with:
```ruby
expect(subject.messages.find { |m| m.system? }).to be_present
subject.instructions = nil
expect(subject.messages).to be_empty
expect(subject.instructions).to be_nil
expect(subject.messages.find { |m| m.system? }).to_not be_present
```

To make sure that the system message gets dropped.